### DISCUSSION TITLE
What does it mean that  "you cannot undo" the action to "revoke access" to the API?

### DISCUSSION THREAD

**Discussion by bshanks** [community/community#22862] [New to GitHub]
<p>I allowed a third-party service (Gitter) to access my account via the API. I figure it’s a good idea to periodically houseclean and disable access for services i’m not currently using. However, when i click the ‘Revoke access’ button in my account for the Gitter app, i get a scary warning:</p>
<p>"</p>
<p>Are you sure you want to revoke authorization?</p>
<p>Gitter Public Repo Access will no longer be able to access the GitHub API. You cannot undo this action.<br>
"</p>
<p>i thought the action of denying access would be innocuous, but GitHub seems to be trying to warn me that i am about to do something dangerous. What does it mean “you cannot undo this action”? Is it a bad idea to be denying access to third-party services that one intends to continue using in the future, even if one doesn’t intend to use the service for awhile? I was hoping that the next time i attempt to sign in with the third-party service (Gitter), it will be able to re-request access. Is this not the case? Is GitHub telling me that Gitter will be forever blocked from accessing my account and i will not have a chance to reauthorize it later?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>It means that you will have to go through whatever process that the application has for requesting access again. There is no simple 1-click “turn this back on again” button.</p>
